nixos/modules/tasks/filesystems/zfs.nix

```diff
@@ -227,10 +227,26 @@ in
       modulePackage = lib.mkOption {
         internal = true; # It is supposed to be selected automatically, but can be overridden by expert users.
         default = selectModulePackage cfgZfs.package;
+        apply = pkg: pkg.override { inherit (cfgZfs) allowNonLTSKernels; };
         type = lib.types.package;
         description = "Configured ZFS kernel module package.";
       };
 
+      allowNonLTSKernels = lib.mkEnableOption ''
+        usage of ZFS with non-LTS kernels. This is likely to break in most cases
+        and therefore not a supported. This option exists as a convenience for
+        those who wish to run their systems in an unsupported manner.
+
+        Please see the evaluation warning of the ZFS package when using it in
+        such a configuration for more information. If you have never seen such a
+        warning, this option very likely doesn't apply to you.
+
+        :::{.warning}
+        This is explicitly *unsupported*. Do not create issues regarding ZFS
+        kernel support if you have enabled this option.
```

Comment:
Imo this part also needs toning down. At least the grammatical error in the second sentence should be fixed.

Reply:
Yup, forgot to adjust this actually ^^'